awssdk/src/main/scala/Client.scala

```diff
@@ -34,17 +34,17 @@ trait Client[F[_]] {
   /**
     * Get a single value from a table by partition key P.
     */
-  def get[U: Decoder, P: Encoder](
-    tableName: String,
+  def get[P: Encoder, U: Decoder](
+    table: Table,
```

Comment:
The user needs to create a Table now which can be a bit cumbersome:

```      
val partitionKey = Key("id", ScalarAttributeType.S)
val table = Table(tableName, partitionKey, None)
```

In particular the atrributeType of the partition key might not be immediately obvious. Some helper method that assumes it is String perhaps?

Reply:
It is a good point, the current implementation uses `Table` for the sake of reusing model but it doesn't check that the `Encoder` for the partition key/sort key matches the type defined by the `Table`. Also, currently there is nothing stopping the user to call `get[P, S, U]` on `Table` that has no sort key. I have some ideas to build a higher API on top of `Table` so will not address this now.